src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java

```diff
@@ -97,33 +102,35 @@ private String formatTimeToDate(long time) {
         return month.toString() + "/" + day.toString() + "/" + year.toString();
     }
 
-    private void addDARLine(FileWriter darWriter, Document dar, String customContent1, String customContent2, String consentName, String translatedUseRestriction) throws IOException {
-        List<Document> dataSetDetails = dar.get(DarConstants.DATASET_DETAIL, ArrayList.class);
+    private void addDARLine(FileWriter darWriter, DataAccessRequest dar, String customContent1, String customContent2, String consentName, String translatedUseRestriction) throws IOException {
         List<String> datasetNames = new ArrayList<>();
-        if (Objects.nonNull(dataSetDetails)) {
-            for (Document detail : dataSetDetails) {
-                datasetNames.add(StringUtils.defaultString(detail.getString("name")));
-            }
-        }
-        List<Integer> dataSetIds = DarUtil.getIntegerList(dar, DarConstants.DATASET_ID);
+        List<Integer> dataSetIds = new ArrayList<>();
         List<String> dataSetUUIds = new ArrayList<>();
-        if (Objects.nonNull(dataSetIds)) {
-            for (Integer id : dataSetIds) {
-                dataSetUUIds.add(DataSet.parseAliasToIdentifier(id));
+        if (Objects.nonNull(dar) && Objects.nonNull(dar.getData())) {
+            List<DatasetDetailEntry> dataSetDetails = Objects.nonNull(dar.getData().getDatasetDetail()) ? dar.getData().getDatasetDetail() : Collections.emptyList();
+            for (DatasetDetailEntry detail : dataSetDetails) {
+                try {
+                    Integer id = Integer.parseInt(detail.getDatasetId());
+                    dataSetIds.add(id);
+                    dataSetUUIds.add(DataSet.parseAliasToIdentifier(id));
+                } catch (Exception e) {
+                    logger.warn("Invalid Dataset ID");
```

Comment:
In practice, I see this warning repeated over and over again. It would be helpful to know what the invalid dataset id is and what the DAR reference id is as well so they can be cleaned up.